extract-generated-code-doc/src/doc.rs

```diff
@@ -0,0 +1,440 @@
+    let ext_trait = Sections::new_from_trait(x11rb_xproto.get_by_needle("pub trait ConnectionExt"));
+    DOCS.replace("{protocol_header}", &proto_xproto.get_range_by_index(..4))
+        .replace(
+            "{protocol_window}",
+            proto_xproto.get_by_needle("pub type Window"),
+        )
+        .replace(
+            "{protocol_point}",
+            proto_xproto.get_by_needle("pub struct Point"),
+        )
+        .replace(
+            "{protocol_depth}",
+            proto_xproto.get_by_needle("pub struct Depth"),
+        )
+        .replace(
+            "{protocol_backing_store}",
+            proto_xproto.get_by_needle("pub struct BackingStore"),
+        )
+        .replace(
+            "{protocol_config_window}",
+            proto_xproto.get_by_needle("pub struct ConfigWindow"),
+        )
+        .replace(
+            "{protocol_client_message_data}",
+            proto_xproto.get_by_needle("pub struct ClientMessageData"),
+        )
+        .replace(
+            "{protocol_key_press_event}",
+            &shorten_docs(proto_xproto.get_by_needle("pub struct KeyPressEvent")),
+        )
+        .replace(
+            "{protocol_request_error}",
+            proto_xproto.get_by_needle("pub const REQUEST_ERROR"),
+        )
+        .replace(
+            "{protocol_no_operation}",
+            proto_xproto.get_by_needle("pub struct NoOperationRequest"),
+        )
+        .replace(
+            "{x11rb_no_operation}",
+            x11rb_xproto.get_by_needle("pub fn no_operation"),
+        )
+        .replace(
+            "{x11rb_no_operation_trait}",
+            &shorten_docs(ext_trait.get_by_needle("fn no_operation")),
+        )
+        .replace(
+            "{protocol_get_input_focus}",
+            &shorten_docs(proto_xproto.get_by_needle("pub struct GetInputFocusRequest")),
+        )
+        .replace(
+            "{protocol_get_input_focus_reply}",
+            &shorten_docs(proto_xproto.get_by_needle("pub struct GetInputFocusReply")),
+        )
+        .replace(
+            "{x11rb_get_input_focus}",
+            x11rb_xproto.get_by_needle("pub fn get_input_focus"),
+        )
+        .replace(
+            "{x11rb_get_input_focus_trait}",
+            &shorten_docs(ext_trait.get_by_needle("fn get_input_focus")),
+        )
+        .replace(
+            "{protocol_create_window_aux}",
+            &shorten_create_window_aux(proto_xproto.get_by_needle("pub struct CreateWindowAux")),
+        )
+        .replace(
+            "{protocol_create_window_request}",
+            &shorten_docs(proto_xproto.get_by_needle("pub struct CreateWindowRequest")),
+        )
+        .replace(
+            "{x11rb_create_window_request}",
+            &shorten_docs(x11rb_xproto.get_by_needle("pub fn create_window")),
+        )
+        .replace(
+            "{x11rb_create_window_request_trait}",
+            &shorten_docs(ext_trait.get_by_needle("fn create_window")),
+        )
+}
```

Comment:
There are multiple Rust templating libraries that would help with this, we may want to consider using one (but we can leave that for a future PR)

Reply:
I could also refactor this into some `for` loop. Dunno if that makes it much better.

Dunno if a template library is really "worth it" for this simple case.